src/escpos/escpos.py

```diff
@@ -422,6 +453,128 @@ def barcode(
         align_ct=True,
         function_type=None,
         check=True,
+        force_software=False,
+    ):
+        """Print barcode.
+
+        Automatic hardware|software barcode renderer according to the printer capabilities.
+
+        Defaults to hardware barcode and its format types if supported.
+        Automatically switches to software barcode renderer if hardware does not
+        support a barcode type that is supported by software. (e.g. JAN, ISSN, etc.).
+
+        Set force_software=True to force the software renderer according to the profile.
+        Set force_software=graphics|bitImageColumn|bitImageRaster to specify a renderer.
+
+        Ignores caps, special chars and whitespaces in barcode type names.
+        So "EAN13", "ean-13", "Ean_13", "EAN 13" are all accepted.
+
+        :param code: alphanumeric data to be printed as bar code (payload).
+
+        :param bc: barcode format type (EAN13, CODE128, JAN, etc.).
+
+        :param height: barcode module height (in printer dots), has to be between 1 and 255.
+            *default*: 64
+        :type height: int
+
+        :param width: barcode module width (in printer dots), has to be between 2 and 6.
+            *default*: 3
+        :type width: int
+
+        :param pos: text position (ABOVE, BELOW, BOTH, OFF) relative to the barcode
+            (ignored in software renderer).
+            *default*: BELOW
+
+        :param font: select font A or B (ignored in software renderer).
+            *default*: A
+
+        :param align_ct: If *True*, center the barcode.
+            *default*: A
+        :type align_ct: bool
+
+        :param function_type: ESCPOS function type A or B. None to guess it from profile
+            (ignored in software renderer).
+            *default*: None
+
+        :param check: If *True*, checks that the code meets the requirements of the barcode type.
+            *default*: True
+        :type check: bool
+
+        :param force_software: If *True*, force the use of software barcode renderer from profile.
+            If *"graphics", "bitImageColumn" or "bitImageRaster"*, force the use of specific renderer.
+        :type force_software: bool | str
+
+        :raises: :py:exc:`~escpos.exceptions.BarcodeCodeError`,
+                 :py:exc:`~escpos.exceptions.BarcodeTypeError`
+
+        .. tip::
+            Get all supported formats at:
```

Comment:
Maybe replace the broken references with a table containing the full barcode types that are supported, ignoring the render method, is better. And additionally change the label ``tip`` to ``note``?

```
| Supported barcode formats |
| :---: |
| UPCA |
| UPCE |
| EAN13 |
| EAN8 |
| CODE39 |
| ITF |
| NW7 |
| CODABAR |
| CODE93 |
| CODE128 |
| GS1_128 |
| GS1_DATABAR_OMNIDIRECTIONAL | GS1_DATABAR_TRUNCATED |
| GS1_DATABAR_LIMITED |
| GS1_DATABAR_EXPANDED |
| EAN |
| EAN13GUARD |
| EAN14 |
| EAN8GUARD |
| GS1 |
| GTIN |
| ISBN |
| ISBN10 |
| ISBN13 |
| ISSN |
| JAN |
| PZN |
| UPC |
```

Reply:
Sounds good :-)

Reply:
I have thought a bit more about it: how can we keep this list up to date? In the end this changes when python-barcode gets more support.

Please have a look at https://python-escpos--527.org.readthedocs.build/en/527/api/escpos.html#escpos.escpos.Escpos.barcode. If you think this is OK, we could merge it like this, otherwise feel free to add the list as proposed.

Reply:
I think that's great. In fact, that's what I was trying to get from the beginning, but I didn't know how to do it properly.